include/yaclib/coroutine/future_coro_traits.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <yaclib/async/contract.hpp>
+#include <yaclib/async/detail/result_core.hpp>
+#include <yaclib/async/future.hpp>
+#include <yaclib/async/promise.hpp>
+#include <yaclib/util/detail/atomic_counter.hpp>
+#include <yaclib/util/intrusive_ptr.hpp>
+#include <yaclib/util/result.hpp>
+
+#include <exception>
+
+#if defined(__clang__)
+
+#include <experimental/coroutine>
+namespace yaclib_std {
+
+using std::experimental::coroutine_traits;
+
+using suspend_never = std::experimental::suspend_never;
+using suspend_always = std::experimental::suspend_always;
+
+#else
+
+#include <coroutine>
+
+namespace yaclib_std {
+
+using std::coroutine_traits;
+
+using suspend_never = std::suspend_never;
+using suspend_always = std::suspend_always;
+
+#endif
+}  // namespace yaclib_std
+
+template <typename T, typename... Args>
+struct yaclib_std::coroutine_traits<yaclib::Future<T>, Args...> {
+  struct promise_type {
+    yaclib::Future<T, yaclib::StopError> get_return_object() {
+      yaclib::Future<T, yaclib::StopError> ret_obj;
+      std::tie(ret_obj, _internal) = yaclib::MakeContract<T, yaclib::StopError>();
```

Comment:
looks dangerous, why did we construct promise only when get_return_object() was called?
Feel free to use unsafe internal part of API to implement coroutines